### PR TITLE
New: Create option to enable JSXText node type (fixes #266)

### LINKED ISSUE #266: JSXText nodes are mislabeled as Literal nodes
**What version of TypeScript are you using?**

2.3.2

**What version of `typescript-eslint-parser` are you using?**

a294afa8158c9c088521eed72b6745eed302361c

**What code were you trying to parse?**

```jsx
<a> </a>
```

**What did you expect to happen?**

`Program.body[0].expression.children[0]` should be a `JSXText` node.

**What happened?**

`Program.body[0].expression.children[0]` is a `Literal` node.

```js
{ type: 'Literal',
  range: [ 3, 4 ],
  loc: { start: { line: 1, column: 3 }, end: { line: 1, column: 4 } },
  value: ' ',
  raw: ' ' }
```

prettier/prettier#1558

### PR DESCRIPTION
**Pull request by soda0289**
This commit creates a new parser config option `useJSXTextNode` that enables `JSXText` node type instead of `Literal` for strings inside of `JSXElement` nodes. The default is false as currently many eslint rules depend on the `Literal` node type. 

I have moved the JSX tests into its own directory and created a new test runner that can run tests for JSX files with and without the  `useJSXTextNode`  flag enabled.

### CODE REVIEW

File: tests/lib/jsx.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+//------------------------------------------------------------------------------
+
+const JSX_FIXTURES_DIR = "./tests/fixtures/jsx";
+
+const filesWithOutsandingTSIssues = [
+    "jsx/embedded-tags", // https://github.com/Microsoft/TypeScript/issues/7410
+    "jsx/namespaced-attribute-and-value-inserted", // https://github.com/Microsoft/TypeScript/issues/7411
+    "jsx/namespaced-name-and-attribute", // https://github.com/Microsoft/TypeScript/issues/7411
+    "jsx/invalid-namespace-value-with-dots", // https://github.com/Microsoft/TypeScript/issues/7411
+    "jsx/multiple-blank-spaces" // Fixed by #271
```
Comment (JamesHenry) [inline tests/lib/jsx.js]: This was fixed, why has it been re-added as one to skip?

File: lib/convert.js [hunk 2]
```diff
@@ -1613,9 +1613,12 @@ module.exports = function convert(config) {
 
         }
 
-        case SyntaxKind.JsxText:
+        case SyntaxKind.JsxText: {
```
Comment (JamesHenry) [inline lib/convert.js]: Please can you document inline why we are doing this?

Comment (soda0289) [inline lib/convert.js]: Sure. It is only temporary until eslint v5, will make a note.

File: tests/lib/jsx.js (hunk 1 above)
Comment (soda0289) [inline tests/lib/jsx.js]: Once I rebase this change I can remove it.

**Comment by JamesHenry** [comment]
Ah, I just merged the other JSXText PR, so this will need rebasing. Once that's done this is good to go, and I can see it has already been verified in prettier.

**Comment by soda0289** [comment]
@JamesHenry Rebased. Should be good to go in.

**Pull request merged**